core/base/document.php

```diff
@@ -1207,6 +1207,32 @@ public static function on_import_update_dynamic_content( array $config, array $d
 		return $config;
 	}
 
+	/**
+   * Update dynamic settings in the document for import.
+   *
+   * @param array $settings The settings of the document.
+   * @param array $config Import config to update the settings.
+   *
+   * @return array
+   */
+	public function on_import_update_dynamic_settings( array $settings, array $config ): array {
+		$controls = $this->get_controls();
+		$controls_manager = Plugin::$instance->controls_manager;
+
+		foreach ( $settings as $key => $value ) {
+			if ( isset( $controls[ $key ] ) ) {
+				$control = $controls[ $key ];
```

Comment:
Maybe? More readable.
```suggestion
			if ( ! isset( $controls[ $key ] ) ) {
			continue;
			}
			
			$control = $controls[ $key ];
			...
```